logstash-core/src/test/java/org/logstash/common/io/DeadLetterQueueWriterTest.java

```diff
@@ -92,4 +98,33 @@ public void testWrite() throws Exception {
         writer.writeEntry(entry);
         writer.close();
     }
+
+    @Test
+    public void testDoesNotWriteBeyondLimit() throws Exception {
+        DLQEntry entry = new DLQEntry(new Event(), "type", "id", "reason");
+
+        int payloadLength = RECORD_HEADER_SIZE + VERSION_SIZE + entry.serialize().length;
+        long queueLength = payloadLength * 5;
+        DeadLetterQueueWriter writer = new DeadLetterQueueWriter(dir, payloadLength, queueLength);
+
+        for (int i = 0; i < 5; i++)
+            writer.writeEntry(entry);
+
+        long size = Files.list(dir)
+                         .filter(p -> p.toString().endsWith(".log"))
+                            .mapToLong(p -> p.toFile().length())
+                            .sum();
+
+        assertThat(size, is(queueLength));
+
```

Comment:
@robbavey shouldn't we do another `writer.writeEntry` to actually test that the size doesn't increase further?